src/core/transaction.zig

```diff
@@ -254,11 +253,21 @@ pub const Message = struct {
     }
 
     pub fn clone(self: *const Message, allocator: std.mem.Allocator) error{OutOfMemory}!Message {
+        const account_keys = try allocator.dupe(Pubkey, self.account_keys);
+        errdefer allocator.free(account_keys);
+
         const instructions = try allocator.alloc(CompiledInstruction, self.instructions.len);
-        for (instructions, 0..) |*ci, i| ci.* = try self.instructions[i].clone(allocator);
+        errdefer allocator.free(instructions);
+
+        for (instructions, self.instructions, 0..) |*ci, original_ci, i| {
+            errdefer for (instructions[0..i]) |prev_ci| prev_ci.deinit(allocator);
+            ci.* = try original_ci.clone(allocator);
+        }
+        errdefer for (instructions) |instruction| instruction.deinit(allocator);
```

Comment:
What is this `errdefer` guarding against?

Reply:
Originally `account_keys` was duped inline in the struct init, must have just left it when I pulled it out and put it above. Since it's not followed by any error it doesn't actually generate any code - I can remove it if you want, but I don't think it's costing us anything either.

Reply:
Makes sense. How about replacing it with a `errdefer comptime unreachable` which statically proves no errors follow it? That's just a suggestion, if not, we can leave it as is.

Reply:
305708cd091871f78b2b5598cad0d3021afd117e